### ISSUE TITLE
Argument types do not match on nullable field

### ISSUE THREAD

**Issue by PVLopes** [hazzik/DelegateDecompiler#79]
I'm trying to add a default value to a nullable field.
In my original code I actually use a `Enum`, but this sample also fails.

I'm using version 0.18.1 (from NuGet) with EF.

Test sample, `TheGood` works, but `TheBad` fails:
```C#
    public partial class MyTable
    {
        public Nullable<short> MyField { get; set; }

        [NotMapped]
        [Computed]
        public short TheBad {
            get { return MyField.HasValue ? (short)MyField : (short)0; }
        }

        [NotMapped]
        [Computed]
        public short TheGood {
            get { return MyField.HasValue ? (short)0 : (short)1; }
        }
    }
```

```
[ArgumentException: Argument types do not match]
   System.Linq.Expressions.Expression.Condition(Expression test, Expression ifTrue, Expression ifFalse) +4380031
   DelegateDecompiler.Address.Merge(Expression test, Address address1, Address address2, IDictionary`2 map) +230
   DelegateDecompiler.ProcessorState.Merge(Expression test, ProcessorState leftState, ProcessorState rightState) +302
   DelegateDecompiler.<>c__DisplayClass1c.<ConditionalBranch>b__1b() +37
   DelegateDecompiler.Processor.Process() +108
   DelegateDecompiler.Processor.Process(VariableInfo[] locals, IList`1 args, Instruction instruction, Type returnType) +127
   DelegateDecompiler.MethodBodyDecompiler.Decompile() +73
   DelegateDecompiler.<>c__DisplayClass1.<Decompile>b__0(MethodInfo m) +47
   DelegateDecompiler.Cache`2.GetOrAdd(TKey key, Func`2 func) +208
   DelegateDecompiler.DecompileExtensions.Decompile(MethodInfo method) +119
   DelegateDecompiler.DecompileExpressionVisitor.Decompile(MethodInfo method, Expression instance, IList`1 arguments) +38
   DelegateDecompiler.DecompileExpressionVisitor.VisitMember(MemberExpression node) +139
   System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor) +14
   System.Linq.Expressions.ExpressionVisitor.Visit(Expression node) +22
   System.Linq.Expressions.ExpressionVisitor.Visit(ReadOnlyCollection`1 nodes) +58
...
```

**Comment by hazzik** [comment]
@PVLopes please check 0.19.0, I think it might be fixed there.

**Comment by hazzik** [comment]
Fixed in 0.19.0 (added tests to ade8762a0cee4fc9e10f6d69cce4dcc617d5133c)

**Comment by PVLopes** [comment]
Thanks, it's working now.

This is a sample of the code that I use:
```C#
public enum MyEnumState
{
    StateA = 0,
    StateB = 1,
    StateC = 2
}

[NotMapped]
[Computed]
public MyEnumState TheState {
    get { return (MyEnumState)(MyField ?? (short)MyEnumState.StateA);
}
```
